engineering_tools/carma

```diff
@@ -339,6 +339,174 @@ carma__stop() {
     fi
 }
 
+carma__swarm() {
+
+    if [ "$1" = "deploy" ]; then
+
+        echo "Checking Docker Swarm status..."
+        if [ "$(docker info | grep Swarm | sed 's/Swarm: //g')" == " inactive" ]; then
+            echo "Initalizing Docker Swarm..."
+            docker swarm init --advertise-addr 192.168.88.100 # consider making a variable
+        else
+            echo "Docker Swarm active"
+        fi
+
+        # if carma-config-data volume doesn't exist exit and request update
+        if [ "$(docker volume ls | grep carma-config-data | sed 's/local     //g')" ]; then
```

Comment:
Does this work? It's checking to see if the internal command there returns an empty string? I've seen `[[ -z ...` used for that, but I'm not super familiar with `[`

Reply:
This fetches the actual volume name "carma-config-data" because at one point I was checking name strings. Now that I realize I forgot to revert it back to just a grep check I am also shocked it works. Will fix for clarity